openeo/extra/job_management/job_splitting.py

```diff
@@ -0,0 +1,136 @@
+import abc
+import math
+from typing import Dict, List, NamedTuple, Optional, Union
+
+import pyproj
+import shapely
+from shapely.geometry import shape
+from shapely.ops import transform
+
+
+class JobSplittingFailure(Exception):
+    pass
+
+
+# TODO: This function is also defined in openeo-python-driver. But maybe we want to avoid a dependency on openeo-python-driver?
+def reproject_bounding_box(bbox: Dict, from_crs: Optional[str], to_crs: str) -> Dict:
+    """
+    Reproject given bounding box dictionary
+
+    :param bbox: bbox dict with fields "west", "south", "east", "north"
+    :param from_crs: source CRS. Specify `None` to use the "crs" field of input bbox dict
+    :param to_crs: target CRS
+    :return: bbox dict (fields "west", "south", "east", "north", "crs")
+    """
+    box = shapely.geometry.box(bbox["west"], bbox["south"], bbox["east"], bbox["north"])
+    if from_crs is None:
+        from_crs = bbox["crs"]
+    tranformer = pyproj.Transformer.from_crs(crs_from=from_crs, crs_to=to_crs, always_xy=True)
+    reprojected = transform(tranformer.transform, box)
+    return dict(zip(["west", "south", "east", "north"], reprojected.bounds), crs=to_crs)
+
+
+# TODO: This class is also defined in openeo-aggregator. But maybe we want to avoid a dependency on openeo-aggregator?
+class BoundingBox(NamedTuple):
+    """Simple NamedTuple container for a bounding box"""
+
+    west: float
+    south: float
+    east: float
+    north: float
+    crs: str = "EPSG:4326"
+
+    @classmethod
+    def from_dict(cls, d: Dict) -> "BoundingBox":
+        return cls(**{k: d[k] for k in cls._fields if k not in cls._field_defaults or k in d})
+
+    @classmethod
+    def from_polygon(cls, polygon: shapely.geometry.Polygon, projection: Optional[str] = None) -> "BoundingBox":
```

Comment:
why only `Polygon` support and no `MultiPolygon`? I think we can support that here without any effort